node/test/unit/tessel.js

```diff
@@ -1616,3 +1625,373 @@ exports['Tessel.SPI'] = {
     test.done();
   }
 };
+
+exports['Tessel.Wifi'] = {
+  setUp: function(done) {
+    this.Port = sandbox.stub(Tessel, 'Port');
+    this.fsWrite = sandbox.stub(fs, 'writeFile');
+    this.exec = sandbox.stub(childProcess, 'exec', (cmd, callback) => {
+      callback();
+    });
+    this.tessel = new Tessel();
+    done();
+  },
+
+  tearDown: function(done) {
+    Tessel.instance = null;
+    sandbox.restore();
+    done();
+  },
+
+  initialized: function(test) {
+    test.expect(3);
+
+    test.equal(this.tessel.network.wifi.isBusy, false, 'not busy by default');
+    test.equal(this.tessel.network.wifi.isConnected, false, 'not connected by default');
+    test.deepEqual(this.tessel.network.wifi.settings, {}, 'no setings by default');
+
+    test.done();
+  },
+
+  connect: function(test) {
+    test.expect(6);
+
+    var settings = {
+      ssid: 'TestNetwork',
+      password: 'TestPassword',
+      security: 'psk2'
+    };
+    var ip = '192.168.1.101';
+    var network = {
+      ssid: 'TestNetwork',
+      strength: '30/80'
+    };
+
+    this.exec.restore();
+    this.exec = sandbox.stub(childProcess, 'exec', (cmd, callback) => {
+      if (cmd === 'ifconfig wlan0') {
+        callback(null, ip);
+      } else if (cmd === 'ubus call iwinfo info {"device":"wlan0"}') {
+        callback(null, JSON.stringify(network));
+      } else {
+        callback();
+      }
+    });
```

Comment:
Actually, it looks like you don't change `ip` and `network` when using this function so you could just define it all in the `setUp` function.